### DISCUSSION TITLE
abc_size gem that requires rubocop, rubocop-ast and merges part of spec - licensing question

### DISCUSSION THREAD

**Discussion by efurtak** [rubocop/rubocop#10281] [Show and tell]
Hello Team!

My favourite code editor don't have an integrated ABC size measurement, so I've created a gem for that.

It requires `rubocop` and `rubocop-ast`, just to not reinvent the wheel.
It may look like a tugboat near two tankers, but it works so far ;-)

In the code I'm invoking methods from above-mentioned repositories. I've also merged a big part of one of Your spec into mine, to test if ABC is calculated properly for nodes selected from abstract syntax tree.

I've read licenses of Your repositories - those are MIT's and allows a lot without limitation, eg. modifying and merging of the "software".

But English is not my native language, and I would like to make sure:
Is publishing the code as a gem, as a public github repository, would only require the same MIT license but with my name? Would that be enough? If not, what would be?

Link to the main file: https://github.com/efurtak/abc_size/blob/main/lib/abc_size.rb
Link to the spec: https://github.com/efurtak/abc_size/blob/main/spec/abc_size/calculator_spec.rb

And the link to the whole repository: https://github.com/efurtak/abc_size

**Comment by marcandre** [comment]
You are fine.

I believe that you are simply using and referring to `rubocop`, so the license does not even apply.

If instead you copied a substantial portion of `rubocop`'s code, then it would apply (allowed, and you would have to include the license too)

You may want to check [wikipedia](https://en.wikipedia.org/wiki/MIT_License) and select your native language.

Note: IANAL

Good luck

**Reply by efurtak** [reply]
I've checked wikipedia, it's a good hint.

But in case of any dispute it's not counted at all, as this is only a translation.

Referring to what you wrote, key words here would be `substantial portion` and `copy`.
And those are cases that should include copyright notice and permission notice.

Now I understand it better. Thank You!
